package arrays;

import java.util.Arrays;

public class ProductofArrayExceptSelf {
    public static void main(String[] args) {
        int[] nums={1,2,3,4};
        System.out.println(Arrays.toString(product(nums)));
    }
    static int[] product(int[] nums){
        int n = nums.length;
        int[] ans = new int[n];

        int prod = 1;
        for(int i = 0;i < n;i++){
            ans[i] = prod;
            prod *= nums[i];

        }
        int sub = 1;
        for(int i = n-1;i >= 0;i--){
            ans[i] *= sub;
            sub *= nums[i];
        }
        return ans;
    }
}
